tutorial: Migrate to Stargate Remove Launchpad as version name (#604)
cosmos/sdk-tutorials: launchpad-to-stargate/tutorial/02-module-migration.md

---
# Module migration

> In this section, we will be migrating the `pofe` module from our v0.39 application.

Reviewer: ```suggestion
In this section, you migrate the `pofe` module from your v0.39 application.
```

## Module differences

> Before proceeding with the migration, it is important to note the differences between v0.39 and v0.40 Stargate modules. Below is an overview of the differences when scaffolding a module with Starport -

Reviewer: ```suggestion
Before you start the migration, review the differences between v0.39 and v0.40 and later Stargate modules. This overview shows the differences for modules that are scaffolded with Starport:
```

> In theory, you **could** copy a module over from your existing application and make relevant changes based off [this](https://docs.cosmos.network/master/migrations/app_and_modules.html) document - if that is your preferred method, of migrating your modules, then you are better off following the aforementioned documentation.

Reviewer: ```suggestion
In theory, you **could** copy a module from your existing application and make relevant changes based on the information in the [App and Modules Migration](https://docs.cosmos.network/v0.42/migrations/app_and_modules.html#app-and-modules-migration) documentation. 
```

> Instead, we will be using Starport to genrate fresh Stargate-compatible files and migrate the logic from our v0.39 application. Note that we will also have to change some of the logic to make it Stargate compatible.

Reviewer: ```suggestion
This tutorial uses Starport to generate fresh Stargate-compatible files and migrate the logic from a v0.39 application. You will also change some of the logic to make your application Stargate compatible.
```

> Essentially, we need to inspect the `Claim` type that is defined in our v0.39 application.

Reviewer: ```suggestion
The first step is to inspect the `Claim` type that is defined in your v0.39 application.
```

## Protobuf and gRPC

Compared to your v0.39 application, the key differences in type scaffolding is the addition of `proto/pofe/claim.proto`, `x/pofe/keeper/grpc_query_claim.go`, and `x/pofe/types/query_pb.go` files.

> If you take a look inside the `claim.proto` file, you will see the following file which contains transaction (`tx`) messages. We can always assume that these transaction messages will mutate state.

Reviewer: ```suggestion
When you look inside the `claim.proto` file, you see the following file that contains transaction (`tx`) messages. It is safe to assume that these transaction messages mutate state.
```

## Migrating your own application

> Throughout this tutorial, we have shown you how to migrate your application from v0.39 to v0.40+ Stargate with the help of Starport, which will always generate the most up-to-date Cosmos SDK boilerplate code. One might be expectd to migrate their application manually by following the documentation, which is still a viable options, albeit less efficient. 

Reviewer: ```suggestion
Throughout this tutorial, you learned how to use Starport to migrate your application from v0.39 to v0.40 and later Stargate. Starport always generates the most up-to-date Cosmos SDK boilerplate code. 
```

> The approach we used in this tutorial minimizes the changes we had to make when migrating the application - we only had to migrate the custom logic that was implemented, rather than modifying each individual file that needed to be updated. Having boilerplate code also allows us to reference it as an example when examining the fundamental changes within the application itself when making the necessary changes.

Reviewer: ```suggestion
Although it is possible to manually change your application code, this tutorial minimizes the changes required to migrate an application. Instead of manually modifying each file that required updates, you migrated only the custom logic that was implemented. The generated boilerplate code also serves as a useful reference to examine the fundamental changes within the application.
```
